### PR TITLE
Update `protostar.toml` structure. Refactor installation_dir and versioning. E2E.

### PR DESCRIPTION
**Pull request by kasperski95**
- [x] update docs

## Changes
- removed unused fields from `protostar.config`
- fixed typo `DeployedContact` -> `DeployedContract`
- ~added `NEXT_UNSUPPORTED_PROTOSTAR_CONFIG_VERSION`~
- updated docs

```toml
["protostar.config"]
protostar_version = "0.1.0"

["protostar.project"]
libs_path = "lib"

["protostar.contracts"]
main = [
  "./src/main.cairo",
]
```

- added `ProtostarDirectory` and `VersionManager`
- refactored code
- updated e2e tests to run on a fresh protostar build
- updated package management commands use Project

### CODE REVIEW

**Review by MaksymilianDemitraszek — CHANGES_REQUESTED** [review]
LGTM, + the protostar_version changes we discussed

File: src/utils/config/project.py [hunk 1]
```diff
@@ -1,29 +1,36 @@
...
 import tomli_w
+from packaging import version
 
 from src.commands.test.utils import collect_immediate_subdirectories
+from src.config import NEXT_UNSUPPORTED_PROTOSTAR_CONFIG_VERSION
+from src.protostar_exception import ProtostarException
 
 
 class NoProtostarProjectFoundError(Exception):
     pass
```
[the remark this answers is not in the document]
Comment (kasperski95) [reply src/utils/config/project.py:16]: I think `ProtostarException` was introduced later. I'll change that.

File: src/utils/protostar_directory.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import tomli
+from packaging import version
+from packaging.version import LegacyVersion
+from packaging.version import Version as PackagingVersion
+
+from src.protostar_exception import ProtostarException
+
+
+class ProtostarDirectory:
+    class ProtostarNotInstalledException(ProtostarException):
```
Comment (MaksymilianDemitraszek) [inline src/utils/protostar_directory.py:13]: Are we using it anywhere?

File: src/utils/config/project.py [hunk 3]
```diff
@@ -1,39 +1,46 @@
...
-    @classmethod
-    def get_current(cls):
-        return cls()
-
-    def __init__(self, project_root: Optional[Path] = None):
+    def __init__(
+        self, version_manager: VersionManager, project_root: Optional[Path] = None
+    ):
         self.project_root = project_root or Path()
         self._config = None
```
Comment (MaksymilianDemitraszek) [inline src/utils/config/project.py:41]: I would name it `_project_config`

File: src/utils/config/project.py [hunk 4]
```diff
@@ -79,8 +89,40 @@ def load_config(self) -> "ProjectConfig":
...
+                **parsed_config["protostar.project"],
                 "contracts": parsed_config["protostar.contracts"],
             }
             self._config = ProjectConfig(**flat_config)
+
+            self._protostar_config = ProtostarConfig(
+                **parsed_config["protostar.config"],
+            )
+
+            config_protostar_version = version.parse(
```
Comment (MaksymilianDemitraszek) [inline src/utils/config/project.py:101]:
```suggestion
            config_protostar_version = self._version_manager.parse(
```

File: src/utils/protostar_directory.py (hunk 2 above)
Comment (MaksymilianDemitraszek) [reply src/utils/protostar_directory.py:13]: In which scenario user could run protostar without protostar installed? Because Im not sure

**Review by MaksymilianDemitraszek — CHANGES_REQUESTED** [review]
Lots of amazing improvements here! Some nitpicks in comments

File: src/utils/protostar_directory.py (hunk 2 above)
Comment (kasperski95) [reply src/utils/protostar_directory.py:13]: This is a leftover. Previously, I used `os.which('protostar')` to find the installation directory.

**Review by MaksymilianDemitraszek — APPROVED** [review]

**Pull request merged**